lib/ansible/modules/windows/win_domain.ps1

```diff
@@ -51,31 +62,41 @@ Catch { }
 If(-not $forest) {
     $result.changed = $true
 
-    If(-not $check_mode) {
-        $sm_cred = ConvertTo-SecureString $safe_mode_admin_password -AsPlainText -Force
-
-        $install_forest_args = @{
-            DomainName=$dns_domain_name;
-            SafeModeAdministratorPassword=$sm_cred;
-            Confirm=$false;
-            SkipPreChecks=$true;
-            InstallDns=$true;
-            NoRebootOnCompletion=$true;
-        }
-        if ($database_path) {
-            $install_forest_args.DatabasePath = $database_path
-        }
-        if ($sysvol_path) {
-            $install_forest_args.SysvolPath = $sysvol_path
-        }
-        if ($domain_netbios_name) {
-            $install_forest_args.DomainNetBiosName = $domain_netbios_name
-        }
-        
-        $iaf = Install-ADDSForest @install_forest_args
+    $sm_cred = ConvertTo-SecureString $safe_mode_admin_password -AsPlainText -Force
+
+    $install_params = @{
+        DomainName=$dns_domain_name;
+        SafeModeAdministratorPassword=$sm_cred;
+        Confirm=$false;
+        SkipPreChecks=$true;
+        InstallDns=$true;
+        NoRebootOnCompletion=$true;
+        WhatIf=$check_mode;
```

Comment:
We will want to double check `-WhatIf` actually works on the `Install-ADDSForest` cmdlet. There are a few cases where I've seen this get ignored (win_share) and the AD cmdlets don't instill confidence in me.

Reply:
Hmm, I expected -WhatIf to always work, especially since there doesn't seem to be much logic to it (as we hoped it would). And I see we don't have integration tests for **win_domain**.

Reply:
Same but I've been bitten once which has forced me to check things where we didn't use the behaviour before. I would be surprised if it didn't honour it but still something we need to check considering there are no tests.

> And I see we don't have integration tests for win_domain.

Yep that's unfortunate, there's probably little we can test but there should still be some smoke tests to make sure the module is syntactically correct.